src/pages/home/report/ReportActionsList.js

```diff
@@ -83,35 +91,133 @@ function keyExtractor(item) {
     return item.reportActionID;
 }
 
-function ReportActionsList(props) {
+function isUnreadMsg(message, lastRead) {
+    return message && lastRead && message.created && lastRead < message.created;
+}
+
+function ReportActionsList({
+    report,
+    sortedReportActions,
+    windowHeight,
+    onScroll,
+    mostRecentIOUReportActionID,
+    isSmallScreenWidth,
+    personalDetailsList,
+    currentUserPersonalDetails,
+    loadMoreChats,
+    onLayout,
+}) {
     const reportScrollManager = useReportScrollManager();
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
     const opacity = useSharedValue(0);
+    const userActiveSince = useRef(null);
+    const currentUnreadMarker = useRef(null);
+    const scrollingVerticalOffset = useRef(0);
+    const readActionSkipped = useRef(false);
+    const [messageManuallyMarked, setMessageManuallyMarked] = useState(false);
+    const [isFloatingMessageCounterVisible, setIsFloatingMessageCounterVisible] = useState(false);
+    const [reportActionSize, setReportActionSize] = useState(sortedReportActions.length);
     const animatedStyles = useAnimatedStyle(() => ({
         opacity: opacity.value,
     }));
+
     useEffect(() => {
         opacity.value = withTiming(1, {duration: 100});
     }, [opacity]);
     const [skeletonViewHeight, setSkeletonViewHeight] = useState(0);
 
-    const windowHeight = props.windowHeight;
+    useEffect(() => {
+        // If the reportID changes, we reset the userActiveSince to null, we need to do it because
+        // the parent component is sending the previous reportID even when the user isn't activily
+        // into the report
+        if (userActiveSince.current && prevReportID && prevReportID !== report.reportID) {
+            userActiveSince.current = null;
+        } else {
+            userActiveSince.current = DateUtils.getDBTime();
+        }
+        prevReportID = report.reportID;
+    }, [report.reportID]);
```

Comment:
We run into some weird issues trying to use `useRef` that was getting initialized cross-rerenders (when it shouldn't), couldn't find the root cause, will require more time, but seems it is related to Onyx and Navigation. So we have to move out the `prevReportID` to keep track of the last report id. Seems that prev refactor they [do the same](https://github.com/Expensify/App/pull/23729/files)
I need to check the `userActiveSince.current` isn't null to clean it, because Onyx sends the old report ID for a couple of frames, without that check the `useActiveSince` will be updated when it shouldn't